dataset.py

```diff
@@ -5,135 +5,212 @@
-          row["month_of_year"] = month
-          row["lon"] = sample_x
-          row["lat"] = sample_y
-          samples_collected += 1
-
-        except ValueError as e:
-          # masked and out-of-bounds coordinates
-          # note that if sample_radius is zero, we know that looping again
-          # will still result in a ValueError. In this case, we trigger the
-          # loop to finish by incrementing the end condition (samples_collected)
-          # In that case, we end up ignoring the point.
-          # This can also happen with a small positive radius, but one whose
-          # coords return all masked/out of bounds, but this is less likely.
-          if sample_radius == 0:
-            samples_collected += 1
-          continue
-        for key, value in row.items():
-          feature_values[key].append(value)
-
-  samples = pd.DataFrame(feature_values)
-
-  if not monthly:
-    samples.drop("month_of_year", axis=1, inplace=True)
-
-  return samples
-
-def load_sample_data(reference_csv_filename: str) -> pd.DataFrame:
-  """This method loads reference sample data from a CSV, determines the unique
-locations in that CSV and then calls the existing method
-gen_tabular_dataset_with_coords to enrich those locations with features for
-"rh", "temp", "vpd", and  "atmosphere_oxygen_ratio".
-Finally, it joins the CSV provided d18O means values from the CSV with the
-per-loc features"""
+@dataclass
+class DatasetGeographicPartitions:
+    '''
+    Describes the bounds of a geographic area within a dataset.
+    '''
+    min_longitude: float
+    max_longitude: float
+    min_lattitude: float
+    max_lattitude: float
+
+
+_TRAIN_FIXED_BOUNDS = DatasetGeographicPartitions(
+    min_longitude=-62.5,
+    max_longitude=float('inf'),
+    min_lattitude=-5,
+    max_lattitude=float('inf'),
+)
+_VALIDATION_FIXED_BOUNDS = DatasetGeographicPartitions(
+    min_longitude=float('-inf'),
+    max_longitude=-62.5,
+    min_lattitude=-5,
+    max_lattitude=float('inf')
+)
+_TEST_FIXED_BOUNDS = DatasetGeographicPartitions(
+    min_longitude=float('-inf'),
+    max_longitude=float('inf'),
+    min_lattitude=float('-inf'),
+    max_lattitude=-5
+)
+_TRAIN_VALIDATION_TEST_BOUNDS = [
+    _TRAIN_FIXED_BOUNDS, _VALIDATION_FIXED_BOUNDS, _TEST_FIXED_BOUNDS]
+
+
+def _partition_data_fixed(sample_data: pd.DataFrame,
+                         train_validation_test_bounds: list[DatasetGeographicPartitions]) -> PartitionedDataset:
+    '''
+    Return data split between the fixed rectangle train_validation_test_bounds
+    of lattitude and longitude for each of the rows in sample_data. Ranges of partitions are [min, max).
+    '''
+    train_bounds = train_validation_test_bounds[0]
+    validation_bounds = train_validation_test_bounds[1]
+    test_bounds = train_validation_test_bounds[2]
+
+    train_data = sample_data[
+        (sample_data['lat'] >= train_bounds.min_lattitude) & (sample_data['long'] >= train_bounds.min_longitude) &
```

Comment:
nit: Ideally we'd have constants to use for our well known column names.

Reply:
I would agree, but from previous experience the bounds change from dataset to dataset (even if we're accumulating).

Reply:
i just mean 'lat'